blocks/api/parser.js

```diff
@@ -281,7 +281,8 @@ export function createBlockWithFallback( blockNode ) {
 
 		if ( attributesParsedWithDeprecatedVersion ) {
 			block.isValid = true;
-			block.attributes = attributesParsedWithDeprecatedVersion;
+			block.attributes = omit( attributesParsedWithDeprecatedVersion, '_innerBlocks' );
+			block.innerBlocks = attributesParsedWithDeprecatedVersion._innerBlocks || block.innerBlocks;
```

Comment:
This is ugly, **but** I didn't want to change the deprecation API for this, and I suspect this will be a very rare thing to do. Might still want to change it though.

Reply:
A similar issue is being resolved in #5932 Let's try to find the best approach there.

Reply:
Rebased.